R/aggregate.R

```diff
@@ -166,8 +178,9 @@ agg_climate_data <- function(year, data_raster, climate_var, daily_agg, trans =
   keycols = c("x", "y")
   data.table::setkeyv(clim_dt, keycols)
 
+
   # Keyed merge on the x/y column
-  merged_dt <- clim_dt[weights_dt] #cols: x, y, date, value cols 1:k, poly_id, w_area, weight (if weights = T)
+  merged_dt <- clim_dt[weights_dt, allow.cartesian = TRUE] #cols: x, y, date, value cols 1:k, poly_id, w_area, weight (if weights = T)
```

Comment:
@Sara or @anna can you confirm this makes sense? I understood it at some point but now I'm concerned about this kind of merge being unreliable.

Reply:
@tcarleton I think this is required, not in all cases, but without there are times where this merge doesn't work because  the resulting table is basically longer than expected and the function breaks. Just to hopefully quickly go over the issue - `clim_dt` has a row for every lat/lon/date with the associated climate parameter value. The `weights_dt` has lat/lon/polygon id/weights. The two tables are joined on the lat/lon column. The issue is sometimes in the `weights_dt` there are multiple rows for the same lat/lon if that grid cell overlaps with multiple polygons. So there could be 6 rows for the same lat/lon all with different polygon ids and associated weights. This means the join is adding 6 rows for every lat/lon/date combination in `clim_dt` resulting in a table that is larger than the sum of rows in `clim_dt` + `weights_dt` which is triggering the error. `allow.cartesian` just allows for these joins where the output number of rows is greater than the sum of the number of rows in each data table.